### PR TITLE
Added an item with github source authentication to the dotnet CLI in the instructions for GitHub Packages

### PR DESCRIPTION
**Pull request by Falcion**
### Why:

The article itself, although contains information on how to authenticate `github` source for NuGet integration with GitHub Packages still indicate GitHub Actions in the title, which can confuse a user who is working with GitHub Packages for the first time and is trying to upload their project manually (again using `dotnet` CLI)

Example of this experience is this discussion: https://github.com/orgs/community/discussions/49186

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Update guide in [Publishing a package](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#publishing-a-package) and [Publishing a package using a nuget.config file](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#publishing-a-package-using-a-nugetconfig-file) and add step about authenticating GitHub Packages for user:

```markdown
1. Authentificate GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name.

    ```shell
    dotnet nuget add source --username OWNER --password ${ secrets.GITHUB_TOKEN } --store-password-in-clear-text --name github "https://nuget.pkg.github.com/OWNER/index.json"
    ```
```
### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

File: content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md [hunk 1]
```diff
@@ -129,6 +129,12 @@ If you don't already have a {% data variables.product.pat_generic %} to use for
    dotnet pack --configuration Release
    ```
 
+1. Authentificate GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name.
```
Comment (thyeggman) [inline content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md:132]:
```suggestion
1. Authenticate to GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name, and `YOUR_GITHUB_PAT` with your personal access token.
```

**Review by thyeggman — COMMENTED** [review]
This step already exists in the "Authenticating to GitHub Packages" section of the documentation, although it is formatted for use in a GitHub Actions workflow.  I think the confusion is understandable, though.

I would suggest removing these two changes and moving the new part up to the above section. Use `YOUR_GITHUB_PAT` to match other examples instead of `${{ secrets.GITHUB_TOKEN }}`.  Mention that it is a command that can be run directly from the command line using the `dotnet` CLI instead of manually configuring a `nuget.config` file.  Also check the spelling of "Authenticate" 👍

**Comment by Falcion** [comment]
I apologize for the "spammy" commits with fixes and adjustments to your answer.

As was mentioned in the discussion, I removed the “duplicate” authorizations, changed `${{ secrets.GITHUB_TOKEN }}` to `YOUR_GITHUB_PAT`, combined and moved the authorization instructions to the “Publish a package” block, if something else required, I'll be happy to try to fix/finish it.

**Comment by nguyenalex836** [comment]
@Falcion Thank you so much for that last batch of adjustments! 💛 I'll update the branch and get this merged once tests are passing 🍏

**Review by nguyenalex836 — APPROVED** [review]

**Pull request merged**